src/tracer.js

```diff
@@ -254,23 +256,25 @@ export default class Tracer {
     }
 
     /**
-     *  Makes a concrete sampling decision for the ctx span context based on information
-     *  available to it's child span.
-     *
-     *  The ctx context's deferred mask is unset after this decision is made
+     * Reapplies sampling if the parent span isn't sampled
      *
+     * @param parentCtx the parent span context
      * @param ctx the span context
      * @param operationName the operation name for the child
      * @param tags tags to be applied by the sampler
      */
-    processDeferredSampling(ctx: SpanContext, operationName: string, tags: any) {
-        if (ctx.isDeferredSampling) {
+    handleUpsampling(parentCtx: SpanContext, ctx: SpanContext,
+                     operationName: string, tags: any) {
+        if (!this._upsampling.enabled) {
+            return
+        }
+
+        if (!parentCtx.isSampled() && !parentCtx.isDebug()) {
             if (this._sampler.isSampled(operationName, tags)) {
                 ctx._flags |= constants.SAMPLED_MASK;
             } else {
```

Comment:
>ctx._flags &= ~constants.SAMPLED_MASK can reset sampled flag to 0, it is downsampling

We will never reach that line when we encounter a sampled or debug parent because of `!parentCtx.isSampled() && !parentCtx.isDebug()`.